### DISCUSSION TITLE
Cara mengerjakan  bagian test point pada proyek capstone API

### DISCUSSION THREAD

**Discussion by dudupay** [teamalgoritma/community#864] [Q&A] [answered]
Dear Tim Algoritma,

Bagaimana ya cara mangerjakan endpoint pada proyek capstone api? Apakah membuat fungsi def baru di file app.py?

terima kasih

**Comment by IrfanChairurrachman** [comment]
Dear @dudupay , 

Apakah yang dimaksud untuk 3 rubrik terakhir yaitu analytical endpoint? dalam hal ini adalah:
- endpoint statis 
- endpoint dinamis
- endpoint POST

![image](https://github.com/teamalgoritma/community/assets/56690035/365cfd13-19d9-4303-a4eb-89f8779fa0ed)

**Reply by dudupay** [reply]
Hi Ka, 

Iyaa benar. Ini dengan saya Pandu Aditya Kurniawan ya.

Saya juga mendapatkan error di bawah ini.

![image](https://github.com/teamalgoritma/community/assets/165142758/ed568ff3-53a3-4bb9-b65f-8d7d4645fa88)

![image](https://github.com/teamalgoritma/community/assets/165142758/cee7a740-7cc0-4ff2-9e9c-bbfa62c3c4fd)

Mohon bantuannya Ka.

Terima kasih

**Reply by IrfanChairurrachman** [reply]
Jika yang dimaksud benar. Maka betul harus membuat fungsi def baru di file app.py.

Sebagai contoh, pada endpoint statis Anda membuat endpoint untuk return nilai `average_duration` sepeda pada bikeshare. Maka tahapannya:
- Membuat endpoint statis untuk `trips/average_duration` di app.py
- Membuat fungsi query untuk mendapatkan hasil `average_duration` di app.py
- Menjalankan app.py
- Memastikan endpoint yang dibuat berhasil dengan melakukan testing di jupyter notebook pada section "TEST YOUR ENDPOINT" (seperti gambar di bawah). **Jangan lupa untuk mengganti url endpoint sesuai dengan endpoint yang telah dibuat.**

![image](https://github.com/teamalgoritma/community/assets/56690035/61232863-5847-4da9-82b2-54c836fbabaa)

Patut dicatat bahwa `average_duration` ini hanyalah contoh, Anda bisa membuat analisis yang lain seperti mendapatkan jumlah keseluruhan sepeda pada semua stasiun atau analisis lainnya.

Sekian dari kami, 
Semoga membantu

**Reply by IrfanChairurrachman** [reply]
Dear @dudupay ,

Jika kita melihat errornya, maka terdapat pada line 25 dan 61 di `app.py`. Silahkan dikirimkan terlebih dahulu screenshot kode di line tersebut pada `app.py` (khususnya untuk fungsi yang berhubungan dengan `/trips/{id}`) agar kami dapat membantu untuk mencari sumber kesalahannya.

Terima kasih

**Reply by dudupay** [reply]
Ini ya Ka

Sebelumnya bisa tapi skrng malah error

![image](https://github.com/teamalgoritma/community/assets/165142758/49d85382-ba23-43fe-a453-bf8af35580f4)


![image](https://github.com/teamalgoritma/community/assets/165142758/94d4a488-9595-4ebe-81ee-18a6bffb1f58)


Thanks ya Ka

**Reply by dudupay** [reply]
Ka sudah ketemu nih ternyata saya salah di sisi pythonnya di bawah ini

![image](https://github.com/teamalgoritma/community/assets/165142758/cbd9a3a4-b79d-4222-8853-f33ff7884763)

Harusnya ID malah saya station_id

**Reply by IrfanChairurrachman** [reply]
ohiyaa betul Pak, hehe.

oke, baik, terima kasih konfirmasinya Pak. Semangat

**Reply by dudupay** [reply]
Siap Thanks Ka.

Kalau ini masalah koneksi bukan ya ?

![image](https://github.com/teamalgoritma/community/assets/165142758/77bc7806-7562-46c4-b7ed-c1e59e492317)


Saya sudah test end point spt ini kenapa ya Ka?

![image](https://github.com/teamalgoritma/community/assets/165142758/7b73aebb-628a-4898-93a0-5ae43e288ad1)


Terima kasih

**Reply by dudupay** [reply]
Ka @IrfanChairurrachman ,

Mohon bantuannya untuk review query di bawah ini ya. Saya coba error melulu...

![image](https://github.com/teamalgoritma/community/assets/165142758/7a96308f-e08f-4d71-8cb8-870bc2391c80)

![image](https://github.com/teamalgoritma/community/assets/165142758/a87beed9-3844-40c2-a37a-58747e1666cb)


Terima kasih Ka

**Reply by dudupay** [reply]
Ka @IrfanChairurrachman,

Saya tinggal step POST Endpoint. Cara pengerjaannya seperti apa ya?

**Reply by ramdhinafinita** [reply, marked as answer]
Dear @dudupay Untuk pengerjaan capstone secara detil beserta panduannya bisa di tonton pada link videonya, karena tahapan yang akan dikerjakan adalah step POST. Anda bisa merujuk pada video no.8 di drive API. Terimakasih
